lib/psulib_traject/macros/subjects.rb

```diff
@@ -35,24 +35,28 @@ def process_subject_browse_facet(standard_fields:, pst_fields:)
       Traject::MarcExtractor.cached(standard_fields).collect_matching_lines(record) do |field, spec, extractor|
         extracted_subjects = extract_subjects(field, spec, extractor)
         unless extracted_subjects.empty?
-          subjects << PsulibTraject::SubjectHeading.new(extracted_subjects)
+          subjects << PsulibTraject::SubjectHeading.new(extracted_subjects, tag: field.tag)
         end
       end
 
       Traject::MarcExtractor.cached(pst_fields).collect_matching_lines(record) do |field, spec, extractor|
         if field.indicator2 == '7' && (field['2'] || '').match?(/pst/i)
           extracted_subjects = extract_subjects(field, spec, extractor)
           unless extracted_subjects.empty?
-            subjects << PsulibTraject::SubjectHeading.new(extracted_subjects)
+            subjects << PsulibTraject::SubjectHeading.new(extracted_subjects, tag: field.tag)
           end
         end
       end
 
       accumulator.replace(
         subjects
-          .select { |subject| subject.length == subjects.max.length }
-          .map(&:value)
-          .uniq
+          .group_by(&:tag)
+          .map do |_tag, subjectheadings|
+            subjectheadings
+              .select { |subject| subject.length == subjectheadings.max.length }
```

Comment:
I'm not sure I understand how this is working. `subject.length` is now a string. Previously, it was the count of the number of items in the array of headings. We should be returning the subject heading that has the most number of elements, not the one that's the longest.

Reply:
hmm OK, I will fix that, I misunderstood that part 👍🏻

Reply:
I changed the length method to how it was before, I think that kinda did it 🤔